indico/modules/events/client/js/creation.js

```diff
@@ -58,20 +60,39 @@ import {camelizeKeys} from 'indico/utils/case';
       if (mode === 'inheriting') {
         mode = currentCategory.is_protected ? 'inheriting-protected' : 'inheriting-public';
       }
-      const elem = messages.filter('.{0}-protection-message'.format(mode));
-      elem.find('.js-category-title').text(currentCategory.title);
+      const elem = messages.filter(`.${mode}-protection-message`);
+      elem.find('.js-category-title').text(currentCategory && currentCategory.title);
       protectionMessage.html(elem);
     }
 
     options.categoryField.on('indico:categorySelected', (evt, cat) => {
-      if (!currentCategory) {
+      if (!options.listingField.prop('checked')) {
+        options.protectionModeFields.filter('[value=inheriting]').prop('disabled', true);
```

Comment:
I think just disabling "inheriting" is a bad idea; this results in the event being created as a fully public, event.

Some possible options:

- keep the choice, but indicate that it's meaningless for unlisted events
- hide the choice (during creation and management), always create as inheriting, and explicit refuse access in case of an unlisted event if there's no explicit ACL entry
- hide the choice (during creation and management), always create as protected to get the same behavior from the previous bullet point. but then we should probably indicate that the event is protected and ask the user if they want to change it to inheriting when publishing the event to a category

Note that an unlisted event should never be fully public since we do not want to let someone enumerate them by just iterating over IDs and trying to access the event.

Reply:
In this context, I wonder if this message is clear or misleading:

> Unlisted events are not shown in any category **and only the creator can see them.** 

I understand what you mean, but it could also be interpreted as "only the creator is able to access them" which isn't the case of course if the event can be made public.

But with the suggested changes, maybe it's OK - I guess it's obvious that someone who's added to the ACL can access the event... Alternatively, we could change it to something about "only explicitly authorized users" being able to see the event...

Reply:
As discussed, creating them as inheriting and hiding the choices (option 2) seems the more sensible UX. I've implemented that one.